rust-server/Cargo.toml

```
@@ -0,0 +1,16 @@
+[package]
+name = "rust-libp2p-webrtc-server"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+anyhow = "1.0"
+env_logger = "0.10.0"
+futures = "0.3.27"
+libp2p = {version="0.51.1", features = ["identify", "ping", "tokio", "gossipsub", "webrtc", "macros", "kad", "rsa", "ed25519"]}
```

To give https://github.com/libp2p/rust-libp2p/pull/3625 a try you would need to do the following change:

```suggestion
libp2p = {git = "https://github.com/vnermolaev/rust-libp2p.git", branch = "deprecate/gossipsub-close-event", version="0.51.1", features = ["identify", "ping", "tokio", "gossipsub", "webrtc", "macros", "kad", "rsa", "ed25519"]}
```

---

I also gave that branch a try to see if it would resolve https://github.com/libp2p/rust-libp2p/issues/3665 but so far it does not publish messages as expected. The browser gets them from the server, but not vice-versa. I will check back later and retry.